src/cdi_ndi/NDI_TN.cc

```diff
@@ -17,20 +17,48 @@ namespace rtt_cdi_ndi {
 // CONSTRUCTORS
 //----------------------------------------------------------------------------//
 /*!
- * \brief Constructor for NDI reader specific to TN reaction data.
- *
- * This constructor opens an NDI file, navigates to the appropriate data, reads
- * the data into internal buffers, and closes the file. For more details on NDI,
- * see https://xweb.lanl.gov/projects/data/nuclear/ndi/ndi.html
+ * \brief Constructor for NDI reader specific to TN reaction data with provided
+ *        path to gendir file.
  *
  * \param[in] gendir_in path to gendir file
  * \param[in] library_in name of requested NDI data library
  * \param[in] reaction_in name of requested reaction
- * \param[in] mg_form_in choice of multigroup discretization
+ * \param[in] mg_e_bounds_in energy boundaries of multigroup bins (keV)
  */
 NDI_TN::NDI_TN(const std::string &gendir_in, const std::string &library_in,
-               const std::string &reaction_in, const MG_FORM mg_form_in)
-    : NDI_Base(gendir_in, "tn", library_in, reaction_in, mg_form_in) {
+               const std::string &reaction_in,
+               const std::vector<double> mg_e_bounds_in)
+    : NDI_Base("tn", library_in, reaction_in, mg_e_bounds_in) {
+
+  override_gendir_path(gendir_in);
+  load_ndi();
+}
+
+/*!
+ * \brief Constructor for NDI reader specific to TN reaction data using default
+ *        gendir file.
+ *
+ * \param[in] library_in name of requested NDI data library
+ * \param[in] reaction_in name of requested reaction
+ * \param[in] mg_e_bounds_in energy boundaries of multigroup bins (keV)
+ */
+NDI_TN::NDI_TN(const std::string &library_in, const std::string &reaction_in,
```

Comment:
We should consider renaming the class to that it isn't all caps.

Reply:
`NDI_TN` -> `NDI_TNReaction`